File: backend/modules/dtos/response/course.go
package response

import (
	"time"
)

type CourseDTO struct {
	ID          string      `json:"id"`
	Title       string      `json:"title"`
	Slug        string      `json:"slug"`
	Description string      `json:"description"`
	FileID      *string     `json:"file_id,omitempty"`
	File        *FileDTO    `json:"file,omitempty"`
	Modules     []ModuleDTO `json:"modules,omitempty"`
	QuizID      *string     `json:"quiz_id,omitempty"`
	Quiz        *QuizDTO    `json:"quiz,omitempty"`
	CreatedAt   time.Time   `json:"created_at"`
	UpdatedAt   time.Time   `json:"updated_at"`
}

File: frontend/src/types/entities/course.ts
import type { Module, Quiz, File } from "..";

export interface Course {
  id: string;
  title: string;
  slug: string;
  description: string;
  file_id: string | null;
  file: File | null;
  modules: Module[];
  quiz_id: string | null;
  quiz: Quiz | null;
  created_at: Date;
  updated_at: Date;
}
